Lib/typing.py

```diff
@@ -3770,6 +3770,16 @@ def __getattr__(attr):
     elif attr in {"ContextManager", "AsyncContextManager"}:
         import contextlib
         obj = _alias(getattr(contextlib, f"Abstract{attr}"), 2, name=attr, defaults=(bool | None,))
+    elif attr == "_collect_parameters":
+        import warnings
+
+        depr_message = (
+            "The private _collect_parameters function is deprecated and will be"
+            " removed in a future version of Python. Any use of private functions"
+            " is discouraged and may break in the future."
+        )
+        warnings.warn(depr_message, category=DeprecationWarning, stacklevel=1)
```

Comment:
```suggestion
        warnings.warn(depr_message, category=DeprecationWarning, stacklevel=2)
```

Or it literally displays line 3781 of `typing.py` in the DeprecationWarning message as the problematic line when you import it:

```pycon
Python 3.14.0a0 (heads/main:a895756aec, May 10 2024, 14:25:23) [Clang 15.0.0 (clang-1500.3.9.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from typing import _collect_parameters
/Users/alexw/dev/cpython/Lib/typing.py:3781: DeprecationWarning: The private _collect_parameters function is deprecated and will be removed in a future version of Python. Any use of private functions is discouraged and may break in the future.
  warnings.warn(depr_message, category=DeprecationWarning, stacklevel=1)
>>> 
```

You can add a test for this by doing

```py
with self.assertWarns(DeprecationWarning) as cm:
    typing._collect_parameters

self.assertEqual(cm.filename, __file__)

Reply:
Thanks, should have looked at the warning more.